janus.c

```diff
@@ -191,16 +191,42 @@ static gchar *local_ip = NULL;
 gchar *janus_get_local_ip(void) {
 	return local_ip;
 }
-static gchar *public_ip = NULL;
+static GPtrArray *public_ip = NULL;
+guint janus_get_public_ip_count(void) {
+	return public_ip ? public_ip->len : 0;
+}
+gchar *janus_get_public_ip_index(guint index) {
+	if (!public_ip || !public_ip->len) {
+		/* Fallback to the local IP, if we have no public one */
+		return local_ip;
+	}
+	if (index >= public_ip->len) {
+		return NULL;
+	}
+	return g_ptr_array_index(public_ip, index);
+}
 gchar *janus_get_public_ip(void) {
-	/* Fallback to the local IP, if we have no public one */
-	return public_ip ? public_ip : local_ip;
+	return janus_get_public_ip_index(0);
+}
+void janus_set_public_ip_list(GPtrArray *ips) {
+	/* once set do not override */
+	if (public_ip != NULL) {
+		return;
+	}
+	public_ip = g_ptr_array_ref(ips);
 }
-void janus_set_public_ip(const char *ip) {
+static void janus_free_string(gpointer data) {
+	g_free(data);
+}
+void janus_set_public_ip(const gchar *ip) {
 	/* once set do not override */
 	if(ip == NULL || public_ip != NULL)
 		return;
-	public_ip = g_strdup(ip);
+
+	GPtrArray *ips = g_ptr_array_new_with_free_func(janus_free_string);
+	g_ptr_array_add(ips, g_strdup(ip));
+	janus_set_public_ip_list(ips);
+	g_ptr_array_unref(ips);
```

Comment:
Again, I don't understand what this method is supposed to do. I thought it was a convenience method to add IPs as you become aware of them, but it doesn't look like that's the case.

Reply:
This is for backwards compatibility to set a single public IP address (e.g. after getting it from the STUN server).